File: sum-of-squares.cpp
#include <iostream>
#include <map>
using namespace std;

int main()
{
    unsigned int n;
    cin >> n;

    pair<int, int> points[n];
    for (unsigned i = 0; i < n; i++)
    {
        int x, y;
        cin >> x >> y;
        points[i] = make_pair(x, y);
    }
    map<pair<pair<int, int>, pair<int, int>>, unsigned> points_sums;
    unsigned sum = 0;
    for (unsigned i = 0; i < n; i++)
        for (unsigned j = i + 1; j < n; j++)
        {
            if (!points_sums.count(make_pair(points[j], points[i])))
                points_sums[make_pair(points[j], points[i])] = (points[j].first - points[i].first) * (points[j].first - points[i].first) + (points[j].second - points[i].second) * (points[j].second - points[i].second);
            sum += points_sums[make_pair(points[j], points[i])];
        }

    cout << sum;

    return 0;
}

File: sum-of-squares.py
n = int(input())
points = []
for _ in range(n):
    points.append(tuple(map(int, input().split())))

points_dict = {}
for point in points:
    if point not in points_dict.keys():
        points_dict[point] = points.count(point)

sum = 0
for i in range(n):
    for j in range(i + 1, n):
        sum += (
            (points[i][0] - points[j][0]) ** 2 + (points[i][1] - points[j][1]) ** 2
        ) * (points_dict[points[j]] - 1)

print(sum)
